assembly/std/index.ts

```diff
@@ -102,6 +102,94 @@ export function balanceOf(address: string): u64 {
   return env.balanceOf(address);
 }
 
+/**
+ * Check for key in datastore
+ *
+ * @param {StaticArray<u8>} key
+ *
+ * @return {bool} - true if key is present in datastore, false otherwise.
+ */
+export function hasOpKey(key: StaticArray<u8>): bool {
+  let result = env.hasOpKey(key);
+  // From https://doc.rust-lang.org/reference/types/boolean.html &&
+  // https://www.assemblyscript.org/types.html
+  // we can safely cast from u8 to bool
+  return bool(result[0]);
+}
+
+/*
+ * Get data associated with the given key from datastore
+ *
+ * @param {StaticArray<u8>} key
+ *
+ * @return {StaticArray<u8>} - data as a byte array
+ */
+export function getOpData(key: StaticArray<u8>): StaticArray<u8> {
+  return env.getOpData(key);
+}
+
+/*
+ * Get all keys from datastore
+ *
+ * @return {Array<StaticArray<u8>} - a list of key (e.g. a list of bytearray)
+ */
+export function getOpKeys(): Array<StaticArray<u8>> {
+  let keys_ser = env.getOpKeys();
+  return derOpKeys(keys_ser);
+}
+
+export function derOpKeys(keys_ser: StaticArray<u8>): Array<StaticArray<u8>> {
+
+  let MAX_DATASTORE_ENTRY_COUNT = u16.MAX_VALUE;
```

Comment:
maybe it's worth defining a constants.ts for this (and error codes in the future)

Reply:
Done